### PR TITLE
Add session properties to queries

### PR DESCRIPTION
**Pull request by joy-yao**

### CODE REVIEW

**Comment by dain** [comment]
Maybe update the commit message to say "in the verifier"

File: presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java [hunk 1]
```diff
@@ -249,9 +252,15 @@ private QueryResult executeQuery(String url, String username, String password, Q
                 if (explainOnly) {
                     sql = "EXPLAIN " + sql;
                 }
-                try (final ResultSet resultSet = limitedStatement.executeQuery(sql)) {
-                    List<List<Object>> results = limiter.callWithTimeout(getResultSetConverter(resultSet), timeout.toMillis() - stopwatch.elapsed(TimeUnit.MILLISECONDS), TimeUnit.MILLISECONDS, true);
-                    return new QueryResult(State.SUCCESS, null, nanosSince(start), results);
+
+                try {
+                    for (Map.Entry<String, String> entry : sessionProperties.entrySet()) {
+                        limitedStatement.execute("SET SESSION " + entry.getKey() + "='" + entry.getValue() + "'");
```
Comment (electrum) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java]: Do this instead:
```java
connection.unwrap(PrestoConnection.class).setSessionProperty(...);

File: presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java [hunk 2]
```diff
@@ -259,12 +268,12 @@ private QueryResult executeQuery(String url, String username, String password, Q
                     }
                     throw e;
                 }
-                catch (SQLException | VerifierException e) {
```
Comment (electrum) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java]: Why did this move?

File: presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java (hunk 1 above)
Comment (electrum) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java]: I actually don't think `SET SESSION` works with JDBC.

File: presto-verifier/src/main/java/com/facebook/presto/verifier/QueryPairMapper.java [hunk 3]
```diff
@@ -26,12 +32,23 @@
     public QueryPair map(int index, ResultSet rs, StatementContext ctx)
             throws SQLException
     {
+        Map<String, String> sessionProperties = Collections.EMPTY_MAP;
+        String sessionPropertiesStr = rs.getString("session_properties_json");
+        if (sessionPropertiesStr != null) {
+            try {
+                sessionProperties = new ObjectMapper().readValue(sessionPropertiesStr, new TypeReference<HashMap<String, String>>() {});
```
Comment (electrum) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/QueryPairMapper.java]: Use `JsonCodec.mapJsonCodec(String.class, String.class)`

File: presto-verifier/pom.xml [hunk 4]
```diff
@@ -133,7 +137,7 @@
                             <shadedArtifactAttached>true</shadedArtifactAttached>
                             <shadedClassifierName>executable</shadedClassifierName>
                             <transformers>
-                                <transformer implementation="org.apache.maven.plugins.shade.resource.ServicesResourceTransformer" />
+                                <transformer implementation="org.apache.maven.plugins.shade.resource.ServicesResourceTransformer"/>
```
Comment (electrum) [inline presto-verifier/pom.xml]: The space is correct. Code Style -> XML -> Other -> Spaces -> In empty tag

File: presto-verifier/src/main/java/com/facebook/presto/verifier/Query.java [hunk 5]
```diff
@@ -20,14 +22,16 @@
...
 
-    public Query(String catalog, String schema, String query, String username, String password)
+    public Query(String catalog, String schema, String query, String username, String password, Map<String, String> sessionProperties)
     {
         this.catalog = catalog;
         this.schema = schema;
         this.query = clean(query);
         this.username = username;
         this.password = password;
+        this.sessionProperties = sessionProperties;
```
Comment (electrum) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/Query.java]: Defensive copy

File: presto-verifier/src/main/java/com/facebook/presto/verifier/QueryPairMapper.java [hunk 6]
```diff
@@ -26,12 +29,18 @@
     public QueryPair map(int index, ResultSet rs, StatementContext ctx)
             throws SQLException
     {
+        Map<String, String> sessionProperties = Collections.EMPTY_MAP;
+        String sessionPropertiesStr = rs.getString("session_properties_json");
```
Comment (electrum) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/QueryPairMapper.java]: Don't abbreviate variable names

Comment (electrum) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/QueryPairMapper.java]: Can just name this `value`

File: presto-verifier/src/main/java/com/facebook/presto/verifier/QueryPairMapper.java [hunk 7]
```diff
@@ -26,12 +29,18 @@
     public QueryPair map(int index, ResultSet rs, StatementContext ctx)
             throws SQLException
     {
+        Map<String, String> sessionProperties = Collections.EMPTY_MAP;
```
Comment (electrum) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/QueryPairMapper.java]: Use `emptyMap()` or `ImmutableMap.of()`

File: presto-verifier/src/main/java/com/facebook/presto/verifier/QueryPairMapper.java [hunk 8]
```diff
@@ -26,12 +29,18 @@
...
+        String sessionPropertiesStr = rs.getString("session_properties_json");
+        if (sessionPropertiesStr != null) {
+            sessionProperties = JsonCodec.mapJsonCodec(String.class, String.class).fromJson(sessionPropertiesStr);
```
Comment (electrum) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/QueryPairMapper.java]: Make the `JsonCodec` a constant

File: presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java [hunk 9]
```diff
@@ -305,7 +312,8 @@ private void trySetConnectionProperties(Query query, Connection connection)
 
     private Callable<List<List<Object>>> getResultSetConverter(final ResultSet resultSet)
     {
-        return new Callable<List<List<Object>>>() {
+        return new Callable<List<List<Object>>>()
```
Comment (electrum) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java]: Unrelated change

**Comment by electrum** [comment]
Aside from minor comments, looks good
